File: tic-tac-toe-core/src/main/java/by/xxx/pupil/ai/AIPlayer.java
package by.xxx.pupil.ai;


import by.xxx.pupil.model.Board;
import by.xxx.pupil.model.Move;
import by.xxx.pupil.model.Player;

import static by.xxx.pupil.Constants.DEFAULT_AI_TYPE;

public interface AIPlayer {

    Move nextMove(Board board, Player player);

    default Move nextMove(Board board) {
        return nextMove(board, DEFAULT_AI_TYPE);
    }

}

File: tic-tac-toe-core/src/main/kotlin/by/xxx/pupil/ai/AIPlayer.kt
package by.xxx.pupil.ai

import by.xxx.pupil.model.Board
import by.xxx.pupil.model.Move
import by.xxx.pupil.model.Player

interface AIPlayer {

    /**
     * @param board - describes current game state
     * @param player - whose turn to make a move
     * @param properties - additional properties needed for your algorithm
     */
    fun nextMove(board: Board, player: Player, properties: Map<String, Any>): Move

}